### DISCUSSION TITLE
How to resolve issues generate because of deprecated @prisma/cli

### DISCUSSION THREAD

**Discussion by gkoudjou** [prisma/prisma#10888] [Q&A] [answered]
Hello,

In one of my project, i'm using @prisma/cli 2.14.0

But as stated here, it's now deprecated https://github.com/prisma/prisma/releases/tag/2.16.0

I then replace this version in my package with prisma version 2.15.0

BUT,

When i build my application, there are a lot of missing classes.

Here are some errors : 

```
error TS2305: Module '"../../../node_modules/@prisma/client"' 
has no exported member 'InputJsonValue'.

7 import type { InputJsonValue, JsonValue, Prisma } from '@prisma/client';
```

```
- error TS2314: Generic type 'EmailDelegate<GlobalRejectSettings>' requires 1 type argument(s).

863       this.prisma.approvedSubnet as Prisma.EmailDelegate,
```

Did someone encounter theses errors ? How can I solve them please ?

Thanks

**Comment by janpio** [comment]
Are you updating both the CLI and Client to the same version? Did you `prisma generate` after the update?

**Comment by gkoudjou** [comment]
Hello,

Thanks for you answer.

Yes, I generate the client again. The error then occur when i build the whole application.

> npx prisma generate
> Environment variables loaded from .env
> Prisma schema loaded from prisma\schema.prisma
> 
> ✔ Generated Prisma Client (2.30.3) to .\node_modules\@prisma\client in 2.08s
> You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
> ```
> import { PrismaClient } from '@prisma/client'
> const prisma = new PrismaClient()
> ```

I've also just update the package.json like this 

>     "prisma": "^2.30.3",
>     "@prisma/client": "^2.30.3",

[So that all version of Prisma matches](https://github.com/prisma/prisma/issues/10889#issuecomment-1002209690).

Redo `npm i` then `npx prisma generate` but still have exactly the same errors.

Thanks for your help

**Reply by janpio** [reply, marked as answer]
Then I would say this is just an import that needs to be adapted, similar to what is described here: https://github.com/prisma/prisma/issues/5360#issuecomment-804508001 Should be part of one of the release notes you updated over.

**Reply by gkoudjou** [reply]
Thanks a lot. This fixed the issue for InputJSonValue and JSonValue.

But I still get the last one 

> error TS2314: Generic type 'EmailDelegate<GlobalRejectSettings>' requires 1 type argument(s).
> 
> 875         await (dataType as Prisma.EmailDelegate).update({
>                                ~~~~~~~~~~~~~~~~~~~~

I didn't yet find any issue with the solution.

I'm going to read the releases notes.

Thanks again

**Reply by gkoudjou** [reply]
I've found this issue https://github.com/prisma/prisma/issues/10490 and tryied adding `"skipLibCheck": true` to tsconfig.json but still get the error.

```
error TS2314: Generic type 'EmailDelegate<GlobalRejectSettings>' requires 1 type argument(s).

867       this.prisma.apiKey as Prisma.EmailDelegate,
```

**Reply by gkoudjou** [reply]
Continue my journey..

The code i'm working on is from this project and this line specifically https://github.com/staart/api/blob/075b30d5e4aa4d65fea107334c2b6a42314fc6a3/src/modules/auth/auth.service.ts#L859

The author want to to generically call the findMany on many Delegate.

This code is working with @prisma/cli:2.14.0 but failed with the abode error when switching to prisma:2.30.3.

What I've done so far to "fix" it is to change the code like this (see below)

I don't like the cast to unknown. But didn't find a common Delegate class to all prisma CRUD helper.

Could you please provide me with a better solution please ?


```
    for await (const dataType of [
      this.prisma.membership as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.email as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.session as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.approvedSubnet as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.backupCode as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.identity as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.auditLog as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
      this.prisma.apiKey as unknown as Prisma.EmailDelegate<Prisma.RejectOnNotFound>,
    ] as Prisma.EmailDelegate<Prisma.RejectOnNotFound>[]) {
      for await (const item of await (dataType as Prisma.EmailDelegate<Prisma.RejectOnNotFound>).findMany(
```

**Reply by gkoudjou** [reply]
Happy new year everybody !

Do someone have hint about the above problem please ?

Thanks again for your support

**Reply by janpio** [reply]
This sounds pretty unrelated to the original discussion, so you might have more success by creating a new discussion with a better title that describes this problem.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
